src/CMakeLists.txt

```diff
@@ -30,10 +30,6 @@
 # Version 2.8.3 introduces CMakeParseArguments.
 cmake_minimum_required(VERSION 2.8.3)
 
-# Respect the PATH environment variable when searching for compilers.
```

Comment:
We use these lines so CMake respects the default compiler set in the PATH (see https://stackoverflow.com/questions/29902862/how-to-get-cmake-to-use-the-default-compiler-on-system-path). This is important for our automated tests. What is the reason for removing them?

The PR comment says that this is done so the host can decide which compiler to use. This is already possible like this:
```CC=/path/to/gcc CXX=/path/to/g++ ./build.py```

Reply:
The problem is that you force the look-up for g++, when some clients like me rather use clang.
According to https://stackoverflow.com/questions/29902862/how-to-get-cmake-to-use-the-default-compiler-on-system-path, the system already looks up for your default compiler under the names `cc` and `c++`. If that does not work for you, that means that the `PATH` you provide for your tests do not provide these binaries. Show me your automated tests and let's try to fix it.

Reply:
The code here uses `gcc` and `g++` only as a fallback. The first choice is `$ENV{CC}` and `$ENC{CXX}`, so it already uses `CC` and `CXX`. This is the goal of those lines: if `CC` and `CXX` are set, they should be used instead of the system's default compiler. You can use clang with `CC=clang CXX=clang++ ./build.py`.

Our automated tests do this as well: https://github.com/aibasel/downward/blob/main/.github/workflows/ubuntu.yml
The relevant lines are 33 and 34:
```
      CC: ${{ matrix.compiler-version.cc }}
      CXX: ${{ matrix.compiler-version.cxx }}
```

Maybe a better fix would be to add `cc` and `c++` as a secondary choice (after `$ENV{CC}` but before `gcc`) in the `find_program` call?

Reply:
Actually, since this PR is now closed, lets continue the discussion in the issue tracker.